dspace-spring-rest/src/test/java/org/dspace/app/rest/RelationshipRestRepositoryIT.java

```diff
@@ -2161,6 +2162,114 @@ public void putRelationshipOnlyAccessOnOnePublication() throws Exception {
+
+        Item orgUnit1 = ItemBuilder.createItem(context, col3)
+                                   .withTitle("OrgUnit1")
+                                   .withAuthor("Testy, TEst")
+                                   .withIssueDate("2015-01-01")
+                                   .withRelationshipType("OrgUnit")
+                                   .build();
+
+        Item project1 = ItemBuilder.createItem(context, col3)
+                                   .withTitle("Project1")
+                                   .withAuthor("Testy, TEst")
+                                   .withIssueDate("2015-01-01")
+                                   .withRelationshipType("Project")
+                                   .build();
+
+        Item publication = ItemBuilder.createItem(context, col3)
+                                      .withTitle("Publication1")
+                                      .withAuthor("Testy, TEst")
+                                      .withIssueDate("2015-01-01")
+                                      .withRelationshipType("Publication")
+                                      .build();
+
+
+        RelationshipType isOrgUnitOfPersonRelationshipType = relationshipTypeService
+            .findbyTypesAndLabels(context, entityTypeService.findByEntityType(context, "Person"),
+                                  entityTypeService.findByEntityType(context, "OrgUnit"),
+                                  "isOrgUnitOfPerson", "isPersonOfOrgUnit");
+        RelationshipType isOrgUnitOfProjectRelationshipType = relationshipTypeService
+            .findbyTypesAndLabels(context, entityTypeService.findByEntityType(context, "Project"),
+                                  entityTypeService.findByEntityType(context, "OrgUnit"),
+                                  "isOrgUnitOfProject", "isProjectOfOrgUnit");
+        RelationshipType isAuthorOfPublicationRelationshipType = relationshipTypeService
+            .findbyTypesAndLabels(context, entityTypeService.findByEntityType(context, "Publication"),
+                                  entityTypeService.findByEntityType(context, "Person"),
+                                  "isAuthorOfPublication", "isPublicationOfAuthor");
+
+        Relationship relationship1 = RelationshipBuilder
+            .createRelationshipBuilder(context, auhor1, orgUnit1, isOrgUnitOfPersonRelationshipType).build();
+
+        Relationship relationshipAuthorExtra = RelationshipBuilder
+            .createRelationshipBuilder(context, author2, orgUnit1, isOrgUnitOfPersonRelationshipType).build();
+
+        Relationship relationship2 = RelationshipBuilder
+            .createRelationshipBuilder(context, project1, orgUnit1, isOrgUnitOfProjectRelationshipType).build();
+
+        Relationship relationship3 = RelationshipBuilder
+            .createRelationshipBuilder(context, publication, auhor1, isAuthorOfPublicationRelationshipType).build();
+
+        getClient().perform(get("/api/core/relationships/search/byLabel")
```

Comment:
Could you add some inline comments into this long IT method.  It's a bit hard to understand exactly what is going on without a detailed reading of the code.  It'd be nice to give a human-readable explanation of the test data being created, and the two tests performed here against the test data.  Otherwise, this method is hard for others to debug if the test breaks in the future.